### ISSUE TITLE
Dont write default settings sync config to code settings.json

### ISSUE THREAD

**Issue by zzhhbyt1** [shanalikhan/code-settings-sync#513]
**Visual Studio Code Version :** [ 1.21.0 ]
**Code Settings Sync Version :** [ 2.9.0]
**Operating System :** [ Windows 10]
**Occurs On:** [Upload / Download ]
**Proxy Enabled:** [ No ]
**Gist Id:** [ 9d1b34143f9769d9514e22c0e9757147]

![image](https://user-images.githubusercontent.com/11014745/37321585-f2020268-26b3-11e8-98af-5b23dd7b9960.png)

First of all, **THANK YOU** for created such a helpful extension, it saved me many time!

How ever here is a little thing I think can get optimized.
The first part is the settings **I customized** or **should be aware of**.
The second part is the settings which are **equal to default** that the extension automatically **inserted whenever Uploading / Downloading**.
IMO the users shouldn't see and be aware of those "changes" as they **don't make any differences** and only to figure out the global settings were just **polluted** with many unnecessary lines.

I have posted frequently asked questions on 
http://shanalikhan.github.io/2016/11/07/visual-studio-code-frequently-asked-questions.html 
so you may able to resolve the error quickly. 
Please make sure to write the versions and console logged error picture or copied text

**Comment by shanalikhan** [comment]
Nice idea
Will evaluate it

**Comment by shanalikhan** [comment]
1. Last Download
2. Last Upload

IMO are the keys thats best to shift from code user settings to settings sync custom settings file.
Besides that, all the other keys are somehow configurable by user.

**Comment by shanalikhan** [comment]
1. Last Update - Extension auto update this key, its safe to transfer it to Settings Sync configuration
2. Last Download - Extension auto update this key, its safe to transfer it to Settings Sync configuration

Following keys are be global in Settings Sync configuration rather then Code Configuration, as users for Github Enterprise they need to update these settings and than download the settings which replaces them.
1. Host
2. pathPrefix

**Comment by thernstig** [comment]
@shanalikhan Maybe I misunderstood this issue and MR, but maybe you can clarify.

I just downloaded 3.1 and then tried to change my user settings to remove these lines:

```json
    "sync.quietSync": false,
    "sync.removeExtensions": true,
    "sync.syncExtensions": true,
    "sync.forceDownload": false,
```

When I save and upload the settings, those configurations gets added to the bottom to my user settings. Was that not something this issue tried to fix?

**Comment by shanalikhan** [comment]
Yes these settings are gist specific settings, all other keys have been move to a part of global settings for Settings Sync.

**Comment by thernstig** [comment]
@shanalikhan Ok thanks for the clarification. I probably misunderstood the issue then. Since the 4 settings above are set to default values, I would have thought they would not show up in my own user settings.

The reason why I thought so is that any other extensions I have include default settings in the user/workspace settings.json. But I suppose it could be different here for how this plugin works as it syncs settings.

**Comment by ryenus** [comment]
Like @thernstig has pointed out, several settings are added back after uploading, though they have default values:

<img src="https://user-images.githubusercontent.com/610161/44822660-cc3fb380-ac2e-11e8-833e-af1f1d0a7690.png" width="350px" />

**Comment by thernstig** [comment]
@shanalikhan Did you see mine and @ryenus follow-up comment? Is there a reason why default settings are set in the "User Settings" part of VS Code?

**Comment by zzhhbyt1** [comment]
So sorry for being absent for such a long time. And yes, what you guys pointed out is what I was thinking about when I post this issue. Haven't this got fixed now yet?

**Comment by shanalikhan** [comment]
These settings cant move outside from GIST for:
1. These settings are the environment settings, a user can have multiple GIST ( Environment )
2. Other settings that were not related to the user's environment (Single GIST) has been moved to `Global Config` that is shared across multiple environments.

**Comment by zzhhbyt1** [comment]
I mean, if in a gist, the extension can't find these settings they can simply use the factory default behavior right?
And if you are afraid of the default settings being too aggressive like "remove extensions", then it's the default value should be changed and we shouldn't leave a setting there explicitly to warn users. 
Hum, I hope I got your ideas right, if I'm missing something you can explain and clarify my confusion. I'll be grateful!

**Comment by shanalikhan** [comment]
I think I misunderstood your issue back all the time.
What are you asking to **NOT WRITE** sync keys inside the settings.json which have default values/behaviour?
If this is what you are asking let me know I will reopen this issue. :)

**Comment by thernstig** [comment]
@shanalikhan Yes, exactly! (I know I am speaking for @zzhhbyt1 here, but I am pretty sure that is what he means).

That is exectly how it works for any other VS Code extensions I have installed.

**Comment by zzhhbyt1** [comment]
Lol, yeah you guys understand me now finally! XD I shall say in old Chinese guys style:"Sorry for my bad English." XDDDD

**Comment by shanalikhan** [comment]
Yes, i might work on it in few weeks. 
Anyone of you can send me a PR though :smile:

**Comment by shanalikhan** [comment]
Fixed in v3.2 at last.
Let me know if there is any problem
